### PR TITLE
fix: disable network checker if navigator is unavailable

### PR DESCRIPTION
**Pull request by izaaz**
### Summary

Check if navigator is available before using it for network connectivity checks

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

File: packages/analytics-browser/src/plugins/network-connectivity-checker.ts [hunk 1]
```diff
@@ -33,6 +33,14 @@ export const networkConnectivityCheckerPlugin = (): BeforePlugin => {
   };
 
   const setup = async (config: BrowserConfig, amplitude: BrowserClient) => {
+    if (typeof navigator === 'undefined') {
+      config.loggerProvider.debug(
+        'Network connectivity checker plugin is disabled because navigator is not available.',
+      );
+      config.offline = false;
```
Comment (johnf) [inline packages/analytics-browser/src/plugins/network-connectivity-checker.ts:40]: As per your comment in my PR I think you wanted this to be `true`?

**Review by johnf — APPROVED** [review]
Looks good

File: packages/analytics-browser/src/plugins/network-connectivity-checker.ts (hunk 1 above)
Comment (izaaz) [reply packages/analytics-browser/src/plugins/network-connectivity-checker.ts:40]: The comment was to make it false by default. We want to be online if the network connectivity checker plugin was not available.

https://github.com/amplitude/Amplitude-TypeScript/pull/721/files#r1603984757

**Review by Mercy811 — APPROVED** [review]
Thanks @izaaz for fixing when `navigator` is not available when analytics-browser is pulled in the sever side rendering part of a next.js all.

**Pull request merged**